src/test/java/net/openhft/chronicle/queue/TestDeleteQueueFile.java

```diff
@@ -416,6 +416,17 @@ public void run() {
             Jvm.startup().on(TestDeleteQueueFile.class, "Tailer thread terminated: " + direction);
         }
 
+        private DocumentContext readingDocumentWithRetries(ExcerptTailer excerptTailer) {
+            DocumentContext documentContext = null;
+            for (int i = 0; i < 2; i++) {
+                documentContext = excerptTailer.readingDocument();
+                if (documentContext.isPresent()) {
+                    break;
+                }
+            }
+            return documentContext;
+        }
```

Comment:
We need to retry here, but that's OK, that's how people will use the API, the point is to prove with repeated calls to readingDocument you can progress through a queue with files being deleted from it.